lib/similarity_utils.py

```diff
@@ -121,6 +121,14 @@ def print_top_similiar(main_actor: str, names: List[str], values: List[float]) -
 def get_ranking(data: pd.DataFrame, main_actor_id: str, ranking_length: int = 5) -> List[str]:
     """metoda dla ramki danych, id aktora oraz (opcjonalnie) długości rankingu, zwraca listę id aktorów najbardziej
     podobnych do wybranego aktora"""
+    warnings.warn("Jest nowa funkcja get_rankings")
     main_actor = prepare_pandas_row(find_actor(data, main_actor_id))
     ids, similarities = similarity_one_vs_all(data, main_actor)
     return select_top_similiar(ids, similarities, ranking_length)[0]
+
+def get_rankings(data: pd.DataFrame, main_actor: pd.DataFrame, ranking_length: int = 5) -> List[str]:
+    """metoda dla ramki danych, wiersza wybranego aktora oraz (opcjonalnie) długości rankingu, zwraca listę id aktorów najbardziej
+    podobnych do wybranego aktora"""
+    main_actor_row = prepare_pandas_row(main_actor)
```

Comment:
Chyba ustalaliśmy, że similarity ma być liczone na podstawie pythonowych list. Moja propozycja, to żeby również ta funkcja dostawała `main_actor` w postaci listy `List[Any]` zamiast `pd.DataFrame`. Wtedy to konwersja za pomocą `prepare_pandas_row` nie powinno być już potrzebna

Reply:
W modelu chcemy wziąć wszystkich aktorów ze zbioru testowego(ewentualnie jakąś próbę) i znaleźć dla nich rekomendacje a później zastosować metrykę. W związku z tym jeżeli w dane wejściowe będą w postaci pd.DataFrame w mojej ocenie będzie wygodniej. W tym przypadku napisałbym funkcje która będzie używała get_rankings dla każdego kolejnego wiersza ze zbioru testowego(próby). Ale nie umieram się, że musi być to w tej formie. Dane wejściowe w formie Listy wydają się również sensowne ale zbudowanie modelu chyba będzie trudniejsze.

Reply:
Gdzieś i tak trzeba będzie to przekonwertować, bo similarity potrzebuje list. Więc jeżeli tobie będzie wygodniej tak pracować, to ja nie mam nic przeciwko żeby to zostało w postaci pandasowego wiersza